proxy/http/HttpTransact.cc

```diff
@@ -7948,9 +7948,11 @@ HttpTransact::build_response(State *s, HTTPHdr *base_response, HTTPHdr *outgoing
               const char *value;
 
               field = base_response->field_find(fields[i].name, fields[i].len);
-              ink_assert(field != nullptr);
-              value = field->value_get(&len);
-              outgoing_response->value_append(fields[i].name, fields[i].len, value, len, 0);
+              while (field) {
+                value = field->value_get(&len);
+                outgoing_response->value_append(field_name[i], field_len[i], value, len, true);
```

Comment:
This fix can be applicable for Cache-Control header and some other headers but not for every headers.
Please refer to the Note in [RFC7230 Section 3.2.2](https://tools.ietf.org/html/rfc7230#section-3.2.2). At least it's not applicable for Set-Cookie header.

We might be able to have a blacklist / whitelist, but it won't be perfect list. Maybe, just sending all headers separately is a safer way?